build.sc

```diff
@@ -316,19 +305,24 @@ class TerminalModule(val crossScalaVersion: String) extends AmmModule{
 object amm extends Cross[MainModule](fullCrossScalaVersions:_*){
   object util extends Cross[UtilModule](binCrossScalaVersions:_*)
   class UtilModule(val crossScalaVersion: String) extends AmmModule{
-    def moduleDeps = Seq(ops())
+    def moduleDeps = Seq()
     def ivyDeps = T{
-      if (scala3Versions.contains(crossScalaVersion))
-        Agg(
-          ivy"com.lihaoyi:pprint_3:${Deps.pprint.dep.version}",
-          ivy"com.lihaoyi:fansi_3:${Deps.fansi.dep.version}",
-          ivy"org.scala-lang:scala3-library_3:$crossScalaVersion"
-        )
-      else
-        Agg(
-          Deps.pprint,
-          Deps.fansi
+      Agg(
+
+        Deps.osLib,
+        Deps.scalaCollectionCompat,
+        Deps.fansi
+      ) ++ (
+        if (scala3Versions.contains(crossScalaVersion))
+          Agg(
+            ivy"com.lihaoyi:pprint_3:${Deps.pprint.dep.version}",
+            ivy"org.scala-lang.modules::scala-collection-compat:${Deps.scalaCollectionCompat.dep.version}",
+            ivy"org.scala-lang:scala3-library_3:$crossScalaVersion"
+          )
+        else Agg(
+          Deps.pprint
         )
+      )
```

Comment:
@alexarchambault In particular i have no idea what this part is doing haha

Reply:
IIRC, `crossScalaVersion` should be like `3.x`, but as `supports3` is false (the default), we compile this module with Scala 2.13. But `crossScalaVersion` like `3.x` implies it's meant to be consumed by Scala 3 modules nonetheless, so we force some dependencies to have a `_3` suffix (and also pull `scala3-library`, that Mill-built libraries don't pull automatically). It's required for dependencies having macros, like pprint (as Scala 2 and 3 macros are different, and pprint doesn't rely on the [Scala 2 & 3 macro mixing stuff](https://docs.scala-lang.org/scala3/guides/migration/tutorial-macro-mixing.html)).

Reply:
The `ivy"org.scala-lang.modules::scala-collection-compat:…"` seems superfluous here, as it doesn't force any suffix, and there's already `Deps.scalaCollectionCompat` a few lines before.